api/common/sharing/validators.go

```diff
@@ -0,0 +1,54 @@
+package sharing
+
+import (
+	"context"
+	"github.com/Peripli/service-manager/pkg/instance_sharing"
+	"github.com/Peripli/service-manager/pkg/log"
+	"github.com/Peripli/service-manager/pkg/query"
+	"github.com/Peripli/service-manager/pkg/types"
+	"github.com/Peripli/service-manager/pkg/util"
+	"github.com/Peripli/service-manager/pkg/web"
+	"github.com/Peripli/service-manager/storage"
+	"github.com/tidwall/gjson"
+)
+
+func ExtractReferenceInstanceID(ctx context.Context, repository storage.Repository, body []byte, tenantIdentifier string, getTenantId func() string) (string, error) {
```

Comment:
almost there, but it is still part of `validators.go`

Reply:
In general, `api/common/sharing` is a good location for us so we can access those functions for both OSB and SMaaP flows.

How about we split it to:
api/common/sharing:
-> validators.go
-> utils.go

Reply:
functions moved to `utils.go`, resolving